### PR TITLE
Mootube java sol

### PR DESCRIPTION
**Pull request by HarryWangATX**
Added Java sol

### CODE REVIEW

**Comment by HarryWangATX** [comment]
@frodakcin i can change vector bool to array if u want as well

**Comment by HarryWangATX** [comment]
^ referring to previous cpp code

**Comment by frodakcin** [comment]
Sure.

**Comment by frodakcin** [comment]
Actually I feel like it would be more helpful to mention in data structure section if it isn't alr

**Comment by frodakcin** [comment]
Because using it is fine for the most part.

**Comment by frodakcin** [comment]
So in other words, if you want to change it you can, but I don't think it's necessary

**Comment by HarryWangATX** [comment]
@frodakcin ill leave it as is then

**Review by bqi343 — APPROVED** [review]
tabs?

**Review by bqi343 — CHANGES_REQUESTED** [review]
whoops wrong button

File: solutions/USACO jan18-mootube-silver.mdx [hunk 1]
```diff
@@ -63,4 +64,105 @@ int main () {
...
+			return null;
+		}
+
+		public int nextInt() { return Integer.parseInt(next()); }
+		public double nextDouble() { return Double.parseDouble(next()); }
+		public long nextLong() { return Long.parseLong(next()); }
+	}
+}
+
+
```
Comment (frodakcin) [inline solutions/USACO jan18-mootube-silver.mdx:163]: Delete empty line

**Review by frodakcin — CHANGES_REQUESTED** [review]
Minor qol formatting change

File: solutions/USACO jan18-mootube-silver.mdx (hunk 1 above)
Comment (frodakcin) [reply solutions/USACO jan18-mootube-silver.mdx:163]: I'm not sure what the standard is when leaving empty lines between ```s and code, but there definitely shouldn't be a double empty line at the bottom and a single empty line on top.

**Review by frodakcin — APPROVED** [review]

File: solutions/USACO jan18-mootube-silver.mdx [hunk 2]
```diff
@@ -63,4 +64,104 @@ int main () {
 ```
 
 </CPPSection>
+
+<JavaSection>
```
Comment (bqi343) [inline solutions/USACO jan18-mootube-silver.mdx:68]: Add `## Java Implementation`

**Review by bqi343 — CHANGES_REQUESTED** [review]

File: solutions/USACO jan18-mootube-silver.mdx [hunk 3]
```diff
@@ -63,4 +64,106 @@ int main () {
...
+			adj[a].add(new Edge(b, c));
+			adj[b].add(new Edge(a, c));
+		}
+
+		for (int i = 0; i < Q; i++) {
+			vist = new boolean[N];
+			S=0;
+			int a;
+			K = io.nextInt();
+			a = io.nextInt();
```
Comment (bqi343) [inline solutions/USACO jan18-mootube-silver.mdx:106]: just do `int a = io.nextInt()`?

File: solutions/USACO jan18-mootube-silver.mdx [hunk 4]
```diff
@@ -63,4 +64,106 @@ int main () {
...
+			b = io.nextInt();
+			c = io.nextInt();
+			adj[a].add(new Edge(b, c));
+			adj[b].add(new Edge(a, c));
+		}
+
+		for (int i = 0; i < Q; i++) {
+			vist = new boolean[N];
+			S=0;
```
Comment (bqi343) [inline solutions/USACO jan18-mootube-silver.mdx:103]: spacing not consistent with how you assign variables elsewhere

**Review by bqi343 — CHANGES_REQUESTED** [review]

**Pull request merged**